### PR TITLE
✨Add function addon.option.set

### PR DESCRIPTION
**Pull request by sinclairpaul**
# Proposed Changes

Add new function addon.option.set to allow changing addon config.

Example usage:

`bashio::addon.option.set "log_level" "warning"` - Sets the `log_level` key to `warning`, will create if doesn't exist, or overwrite
`bashio::addon.option.set "log_level"` - Removes the `log_level` entry from the config

Please go easy on me 😅

### CODE REVIEW

**Comment by sinclairpaul** [comment]
Just after clicking submit, I realise I haven't considered lists, will likely need further testing.

**Comment by sinclairpaul** [comment]
Thinking about this more, it will always push strings into the JSON as it stands.  Thinking to split out the functions, i.e. remove, edit/add string etc.

Not sure it will be easy to deal with arrays, but it's not likely it is needed.

**Comment by ciotlosm** [comment]
I would have thought this would be part of config not addon.

File: lib/addons.sh [hunk 1]
```diff
@@ -487,6 +487,32 @@ function bashio::addon.options() {
...
+      options=$(bashio::addons "${slug}" "addons.${slug}.options" '.options' | \
+      jq --arg value "$value" --arg key "$key" '.[$key]=$value')
+    else
+      options=$(bashio::addons "${slug}" "addons.${slug}.options" '.options' | \
+      jq --arg key "$key" 'del(.[$key])')
+    fi
+    bashio::api.supervisor POST "/addons/${slug}/options" \
+        "{\"options\": ${options}}"
+    bashio::cache.flush_all
+}
```
Comment (frenck) [inline lib/addons.sh:514]: I think will work

```suggestion
function bashio::addon.option.set() {
    local key=${1}
    local value=${2:-"$null"}
    local slug=${3:-'self'}
    local options
    local payload

    bashio::log.trace "${FUNCNAME[0]}" "$@"

	options=$(bashio::addons "${slug}" "addons.${slug}.options"

    if bashio::var.has_value "${value}"; then
      options=$(bashio::jq "${options}" ".[${key}]=\"${value}\"")
    else
      options=$(bashio::jq "${options}" "del(.[${key}])")
    fi
    
    payload=$(bashio::var.json options "${options}")    
    bashio::api.supervisor POST "/addons/${slug}/options" "${payload}"

    bashio::cache.flush_all
}
```

Maybe rename it to simply `bashio::addon.option`? (Although that isn't clear as well, but we don't use `set` anywhere yet... 🤷

Comment (sinclairpaul) [reply lib/addons.sh:514]: Makes sense, will take another look when I have time (as its been a while).

One challenge I believe was everything was posted as a string, which was an issue for Boolean I believe, or does the var.json fix that?

Comment (frenck) [reply lib/addons.sh:514]: Oh right this will cause issues with non-string values 🤔

Comment (sinclairpaul) [reply lib/addons.sh:514]: Was going to look at parsing switches to jq, but then it got more complicated 😉

Comment (frenck) [reply lib/addons.sh:514]: Maybe we need to introduce `bashio::addon.option` (note the non-plural)

Get: `bashio::addon.option <key>`
Set: `bashio::addon.option <key> <value>`

For the value, we can reuse the logic from `bashio::var.json` that has magic to handle other types (`^` as an escape).

**Pull request closed without merging**